x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx

```diff
@@ -105,7 +113,13 @@ export function getTransactionType({
     : TRANSACTION_REQUEST;
 
   // If the default transaction type is not in transactionTypes the first in the list is returned
-  return transactionTypes.includes(defaultTransactionType)
+  const currentTransactionType = transactionTypes.includes(
+    defaultTransactionType
+  )
     ? defaultTransactionType
     : transactionTypes[0];
+
+  // Replace transactionType in the URL in case it is not one of the types returned by the API
+  replace(history, { query: { transactionType: currentTransactionType } });
```

Comment:
Don't we already have a redirect for transaction type somewhere else?

Reply:
Yes, it was on the service overview component and I removed it: https://github.com/elastic/kibana/pull/136569/files#diff-e678f8ccc9c73af4340ebd64ac8df146b06673ae5ec6031958954b8783a5089eL66-L68